ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -459,16 +456,38 @@ public void replicate() {
     } catch (Throwable e) {
       logger.error("Thread name: {} found some error while replicating from remote hosts", threadName, e);
     } finally {
-      replicationMetrics.updateOneCycleReplicationTime(time.milliseconds() - oneRoundStartTimeMs,
-          replicatingFromRemoteColo, datacenterName);
-      if (fastestReplicaFinishTime != null) {
-        replicationMetrics.updateFastestSlowestReplicaPerCycleTimeDifference(
-            time.milliseconds() - fastestReplicaFinishTime, replicatingFromRemoteColo, datacenterName);
-      }
+      emitReplicationCycleMetrics(remoteReplicaGroups, oneRoundStartTimeMs, time.milliseconds());
     }
     maybeSleepAfterReplication(remoteReplicaGroups.isEmpty());
   }
 
+  /**
+   * Emits metrics for
+   * 1. Replication cycle duration
+   * 2. Cumulative time groups are idle in a cycle i.e groups are in DONE state but cycle continues
+   * 3. Cumulative percentage of time per cycle groups are idle
+   * @param groups list of RemoteReplicaGroups
+   * @param roundStartTime start time of replication cycle
+   * @param roundEndTime end time of replication cycle
+   */
+  private void emitReplicationCycleMetrics(List<RemoteReplicaGroup> groups, long roundStartTime, long roundEndTime) {
+    replicationMetrics.updateOneCycleReplicationTime(roundEndTime - roundStartTime, replicatingFromRemoteColo,
+        datacenterName);
+
+    long totalIdleTime = 0;
+    for (RemoteReplicaGroup group : groups) {
+      totalIdleTime = totalIdleTime + group.getFinishTime() - roundStartTime;
+    }
+    replicationMetrics.updateReplicaBatchTotalIdleTimeMs(totalIdleTime, replicatingFromRemoteColo, datacenterName);
+
+    long cumulativeTime = groups.size() * (roundEndTime - roundStartTime);
```

Comment:
cumulative time is the total time for which the loop ran which is just endtime - starttime.